adaptive_ai/cost_test/cost_analysis_10k/cost_analysis_pipeline.py

```
@@ -0,0 +1,347 @@
+        if self.costs_df is None:
+            return
+
+        plt.style.use("ggplot")
+        fig = plt.figure(figsize=(20, 15))
+        gs = fig.add_gridspec(3, 2)
+
+        # 1. Cost Comparison: Adaptive vs Single Models
+        ax1 = fig.add_subplot(gs[0, :])
+        single_model_costs = {
+            model: self.calculate_single_model_cost(model)
+            for model in TOKEN_PRICES.keys()
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)